### PR TITLE
Add debug test cases for control flows and language constructs

### PR DESCRIPTION
**Pull request by praveennadarajah**
## Purpose
This PR will add debug test cases for control flows and language constructs.

Fixes #https://github.com/ballerina-platform/ballerina-lang/issues/25009

### CODE REVIEW

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/ControlFlowDebugTest.java [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+        assertVariable(lamdaChildVariables[0], "a", "ANT", "string");
+
+        // Variable visibility test for Asynchronous function call (Non-blocking calls)
+        assertVariable(variables[9], "v10_future", "future", "future");
+
+        // Variable visibility test for Asynchronous function call child variables
+        Variable[] asyncChildVariables = getChildVariable(variables[9]);
+        Arrays.sort(lamdaChildVariables, compareByName);
+        assertVariable(asyncChildVariables[1], "result", "90", "int");
+    }
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/ControlFlowDebugTest.java:159]: Is there any special reason for doing variable tests in here? We have separate tests for the variable validation right? Sorry If I've missed anything.

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/LanguageConstructDebugTest.java [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+        debugHitInfo = waitForDebugHit(10000);
+
+        Variable[] variables = fetchDebugHitVariables(debugHitInfo.getRight());
+        Arrays.sort(variables, compareByName);
+
+        // Variable visibility test for object method
+        assertVariable(variables[1], "v02_fullName", "John Doe", "string");
+
+        // Variable visibility test for remote object function
+        assertVariable(variables[3], "v04_statusCode", "500", "int");
```
Comment (NipunaRanasinghe) [inline tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/LanguageConstructDebugTest.java:92]: Is there any special reason for doing variable tests in here? We have separate tests for the variable validation right? Sorry If I've missed anything.

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/ControlFlowDebugTest.java (hunk 1 above)
Comment (praveennadarajah) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/ControlFlowDebugTest.java:159]: This is to verify variable name, type and values are properly visible inside control flows. Previous variable visibility tests do not cover this scenario.

File: tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/LanguageConstructDebugTest.java (hunk 2 above)
Comment (praveennadarajah) [reply tests/jballerina-debugger-integration-test/src/test/java/org/ballerinalang/debugger/test/adapter/LanguageConstructDebugTest.java:92]: This is to verify variable name, type and values are properly visible inside language constructs. Previous variable visibility tests do not cover this scenario.

**Pull request merged**